### ISSUE TITLE
Import functionality after Google Universal Analytics shuts down July 1

### ISSUE THREAD

**Issue by andronocean** [Fyrd/caniuse#6609]
Google's Universal Analytics (the standard "version 3" of Google Analytics) [becomes end-of-life on July 1st, 2023](https://support.google.com/analytics/answer/11583528), and will **stop collecting new data for all sites**. Previously collected data will still be accessible for at least 6 months according to Google. After July 1st, Google Analytics 4 will be the only active analytics tool from Google.[^exception]

The CanIUse data import tool only connects to Universal Analytics properties and does not see GA4 properties at all. Since the main purpose of analytics import is to see your _current_ mix of visitor browsers, this will become less and less useful after the retirement date.

Is there any plan to support importing from GA4?

Alternatively, could a generic local import API be created that lets users add data from any analytics source? This could be very beneficial for sites using other analytics providers (Fathom, Matomo, Cloudflare). There have been a few issues proposing this before: #851, #1724 . Maybe this would be the easiest and most flexible path?

[^exception]: Enterprise customers paying for Analytics 360 have another year of working Universal Analytics, until July 1st 2024.

**Comment by Fyrd** [comment]
Yes, will definitely want to do either of those before the end-of-life, thanks for the reminder. Probably a good opportunity to switch to a more generic import for the reasons you mention.

**Comment by DanielHeath** [comment]
I tried writing an importer from cloudflare analytics (which gives you a list of UA strings and how many requests each made), and discovered that the existing importer has a lot of subtle details that are hard work to reproduce.

**Comment by Fyrd** [comment]
GA4 importer is now available! Visit https://caniuse.com/ciu/import then "Google Analytics (GA4)".

Unfortunately there's no quick way to export the data, so I've provided steps to how to set up an "Exploration" that lets you export data. Hopefully not too hard to follow.

If anyone has any trouble with the export steps or runs into issues importing please let me know!

Also I do still intend to include an option for arbitrary sources as suggested. For now the GA4 importer can roughly be used for that purpose, just provide a CSV file that uses the expected headers:
`Browser,Browser version,Operating system,OS version,Device category,Views`
